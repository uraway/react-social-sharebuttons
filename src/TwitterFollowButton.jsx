// @flow
import React, { Component } from 'react';

declare var twttr: any;

type Props = {
  user: string,
  showCount?: boolean,
};

type State = {
  initialized: boolean,
};

export default class TwitterFollowButton extends Component<Props, State> {
  node = null;

  static defaultProps = {
    showCount: true,
  };

  constructor(props: Props) {
    super(props);
    this.state = { initialized: false };
  }

  componentDidMount() {
    const { initialized } = this.state;
    if (initialized) {
      return;
    }

    if (typeof twttr === 'undefined') {
      const twitterbutton = this.node;
      const twitterscript = document.createElement('script');
      twitterscript.src = '//platform.twitter.com/widgets.js';
      twitterscript.async = true;
      twitterscript.id = 'twitter-wjs';
      if (twitterbutton && twitterbutton.parentNode) twitterbutton.parentNode.appendChild(twitterscript);
    } else {
      twttr.widgets.load();
    }

    this.initialized();
  }

  initialized() {
    this.setState({ initialized: true });
  }

  render() {
    const { user, showCount } = this.props;
    return (
      <a
        ref={(node) => (this.node = node)}
        href={`https://twitter.com/${user}`}
        className="twitter-follow-button"
        data-show-count={showCount}
      >
        Follow @{user}
      </a>
    );
  }
}
