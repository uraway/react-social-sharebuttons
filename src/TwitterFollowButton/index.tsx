import { Component, createRef, RefObject } from 'react';

declare let twttr: {
  widgets: {
    load: () => void;
  };
};

type TwitterFollowButtonProps = {
  user: string;
  showCount?: boolean;
};

export default class TwitterFollowButton extends Component<TwitterFollowButtonProps> {
  ref: RefObject<HTMLAnchorElement>;

  static defaultProps = {
    showCount: true,
  };

  constructor(props: TwitterFollowButtonProps) {
    super(props);
    this.ref = createRef();
  }

  componentDidMount(): void {
    if (typeof twttr === 'undefined') {
      const s = document.createElement('script');
      s.src = '//platform.twitter.com/widgets.js';
      s.async = true;
      s.id = 'twitter-wjs';
      if (this.ref.current?.parentNode) this.ref.current.parentNode.appendChild(s);
    } else {
      twttr.widgets.load();
    }
  }

  render(): JSX.Element {
    const { user, showCount } = this.props;
    return (
      <a
        ref={this.ref}
        href={`https://twitter.com/${user}`}
        className="twitter-follow-button"
        data-show-count={showCount}
      >
        Follow @{user}
      </a>
    );
  }
}
