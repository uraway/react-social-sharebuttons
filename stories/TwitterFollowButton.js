import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, boolean, text } from '@storybook/addon-knobs';
import { TwitterFollowButton } from '../dist/index.esm.js';

export default storiesOf('TwitterFollowButton', module)
  .addDecorator(withKnobs)
  .add('configured', () => (
    <TwitterFollowButton user={text('user', 'uraway_')} showCount={boolean('showCount', true)} />
  ));
