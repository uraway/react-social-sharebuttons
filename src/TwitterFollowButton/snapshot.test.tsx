import renderer from 'react-test-renderer';
import TwitterFollowButton from '.';

it('renders correctly with required props', () => {
  const tree = renderer.create(<TwitterFollowButton user="user" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with other props', () => {
  const tree = renderer.create(<TwitterFollowButton user="user" showCount={false} />).toJSON();
  expect(tree).toMatchSnapshot();
});
