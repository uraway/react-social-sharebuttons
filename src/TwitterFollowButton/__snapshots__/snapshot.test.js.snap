// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with other props 1`] = `
<div>
  <a
    className="twitter-follow-button"
    data-show-count={false}
    href="https://twitter.com/user"
  >
    Follow @
    user
  </a>
</div>
`;

exports[`renders correctly with required props 1`] = `
<div>
  <a
    className="twitter-follow-button"
    data-show-count={true}
    href="https://twitter.com/undefined"
  >
    Follow @
  </a>
</div>
`;
